File: src/main/java/com/service/serviceImpl/TransactionCacheHandler.java
package com.service.serviceImpl;

import com.model.Transaction;

import java.math.BigDecimal;
import java.util.*;
import java.util.concurrent.ConcurrentMap;
import java.util.stream.Collectors;

public class TransactionCacheHandler {

    public static List<BigDecimal> getAmountsList(ConcurrentMap<Object, Object> cache) {
        List<BigDecimal> amounts = cache.values()
                                        .stream()
                                        .map(Transaction.class::cast)
                                        .map(Transaction::getAmount)
                                        .collect(Collectors.toList());

        return amounts;
    }
}

File: src/main/kotlin/com/service/serviceImpl/TransactionCacheHandler.kt
package com.service.serviceImpl

import java.util.concurrent.ConcurrentMap
import java.math.BigDecimal
import com.model.Transaction
import mu.KotlinLogging
import org.checkerframework.checker.nullness.qual.NonNull

private val logger = KotlinLogging.logger {}

object TransactionCacheHandler {
    @JvmStatic
    fun getAmountsList(cache: @NonNull ConcurrentMap<@NonNull Any, @NonNull Any>): List<BigDecimal> {
        logger.debug { "getAmountsList parameters: cache: $cache"}

        return cache.values
            .asSequence()
            .map { Transaction::class.java.cast(it) }
            .map { transaction -> transaction.amount }
            .toList()
    }
}